### PR TITLE
Fix GroupsAsSubDomains option for Mesos and Marathon

### LINKED ISSUE #867: Marathon & Mesos providers' GroupsAsSubDomains option broken
Currently, the `GroupsAsSubDomains` option returns a reverse-*sorted* subdomain instead of a reverse-ordered subdomain.

As an example, if the application path is `/d/c/b/a`, the proper subdomain should be `a.b.c.d` (reversed). However, with the current implementation, the subdomain returned is `d.c.b.a` because it always returns in reverse sorted order.

### PR DESCRIPTION
**Pull request by ryanleary**
Bugfix and new tests for the GroupsAsSubDomains bug for Mesos and Marathon providers. Closes #867.

### CODE REVIEW

File: provider/marathon.go [hunk 1]
```diff
@@ -427,7 +426,9 @@ func (provider *Marathon) getFrontendBackend(application marathon.Application) s
 func (provider *Marathon) getSubDomain(name string) string {
 	if provider.GroupsAsSubDomains {
 		splitedName := strings.Split(strings.TrimPrefix(name, "/"), "/")
-		sort.Sort(sort.Reverse(sort.StringSlice(splitedName)))
+		for i, j := 0, len(splitedName)-1; i < j; i, j = i+1, j-1 {
+			splitedName[i], splitedName[j] = splitedName[j], splitedName[i]
+		}
```
Comment (emilevauge) [inline provider/marathon.go]: How is this different from `sort.Reverse(sort.StringSlice(splitedName))` ?

Comment (ryanleary) [inline provider/marathon.go]: sort.Reverse returns the list in reverse SORTED order, not reverse order, per the unit tests added.

Comment (emilevauge) [inline provider/marathon.go]: In fact no, `sort.Reverse(...)` is just a wrapper and does nothing ^^
I would have preferred to get this from std lib but meh ;)

**Comment by emilevauge** [comment]
Maybe we could get this into v1.1.1. Could you change the base of this PR to branch v1.1?

File: provider/marathon.go (hunk 1 above)
Comment (ryanleary) [inline provider/marathon.go]: Sure. Reverse's `Less` implementation just reverses the `Less` implementation of the underlying type. So when we call Sort, it still returns in sorted order. If you have a suggested way to do this using the stdlib, I'd be happy to make that change.

**Comment by ryanleary** [comment]
Included in v1.1.1 would be great. I've had to deploy a custom build in our production environment.

File: provider/marathon.go (hunk 1 above)
Comment (emilevauge) [inline provider/marathon.go]: > If you have a suggested way to do this using the stdlib, I'd be happy to make that change.

Sadly, I don't either ^^

**Comment by ryanleary** [comment]
Do I have to do something to retrigger the CI? Seems stuck.

**Comment by emilevauge** [comment]
@ryanleary Indeed, can you force push maybe ?

**Comment by ryanleary** [comment]
Pulled the reverse code into a function and pushed.

**Comment by Russell-IO** [comment]
LGTM :+1:

**Comment by Russell-IO** [comment]
LGTM :+1:

**Pull request merged**